### DISCUSSION TITLE
In this macro, a value seems to be overwritten unconditionally - why are the initial assignments made?

### DISCUSSION THREAD

**Discussion by vsajip** [congo-cc/congo-parser-generator#136] [Q&A] [answered]
In this macro
https://github.com/congo-cc/congo-parser-generator/blob/430e15d542bbde75750b951a0a1366c8663161d9/src/templates/java/LookaheadRoutines.java.ftl#L135-L158
the lines
https://github.com/congo-cc/congo-parser-generator/blob/430e15d542bbde75750b951a0a1366c8663161d9/src/templates/java/LookaheadRoutines.java.ftl#L147
and
https://github.com/congo-cc/congo-parser-generator/blob/430e15d542bbde75750b951a0a1366c8663161d9/src/templates/java/LookaheadRoutines.java.ftl#L152
seem pointless, because
https://github.com/congo-cc/congo-parser-generator/blob/430e15d542bbde75750b951a0a1366c8663161d9/src/templates/java/LookaheadRoutines.java.ftl#L156
always sets `passedPredicate` to `false` before exiting, and there are no intervening calls - so are the assignments in the `finally` clause only there for when an exception is thrown in the `try`?

**Comment by revusky** [comment, marked as answer]
I think you're misreading the code. The code generated by the `BuildPredicateCode` and/or `BuildScanCode` macros may jump out of the method. And thus, line 152 is not necessarily reached. However, whatever is in the `finally` block is definitely executed.

**Comment by vsajip** [comment]
> macros may jump out of the method

via `return` statements, then, rather than exceptions. Got it.
